src/Extensions/Hosting.Services.Web/Middlewares/ResponseHeadersMiddleware.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Omex.Extensions.Logging;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services.Web.Middlewares
+{
+	/// <summary>
+	/// Adds Omex headers to responces, like MachineId and BuildVersion
+	/// </summary>
+	public class ResponseHeadersMiddleware
+	{
+		internal ResponseHeadersMiddleware(IExecutionContext context) => m_context = context;
+
+		/// <summary>
+		/// Invoke middleware
+		/// </summary>
+		public Task InvokeAsync(HttpContext context)
+		{
+			context.Response.OnStarting(SetResponceHeaders, context.Response);
+			return Task.CompletedTask;
+		}
+
+		private Task SetResponceHeaders(object state)
+		{
+			HttpResponse response = (HttpResponse)state;
+			response.Headers.Add("X-Machine", m_context.MachineId);
+			response.Headers.Add("X-BuildVersion", m_context.BuildVersion); //Renamed from X-OfficeVersion
```

Comment:
This seems strangely related to my other question above, but the naming in the comment has one less "-" than there

Reply:
That's one of the reasons for renaming, client header is `X-Office-Version` and our current version header `X-OfficeVersion` we use them for different purposes and that really confusing.